rust/connlib/tunnel/src/peer.rs

```diff
@@ -192,22 +193,24 @@ impl Default for PacketTransformGateway {
 
 #[derive(Default)]
 pub struct PacketTransformClient {
-    // TODO: we need to refresh the translations ips periodically, just add a timer to resend allow access
     translations: RwLock<BiMap<IpAddr, IpAddr>>,
 }
 
 impl PacketTransformClient {
     pub fn get_or_assign_translation(
         &self,
-        internal_ip: &IpAddr,
         external_ip: &IpAddr,
+        ip_provider: &mut IpProvider,
     ) -> Option<IpAddr> {
         let mut translations = self.translations.write();
         if let Some(internal_ip) = translations.get_by_right(external_ip) {
             return Some(*internal_ip);
         }
-        translations.insert(*internal_ip, *external_ip);
-        Some(*internal_ip)
+
+        let internal_ip = ip_provider.get_ip_for(external_ip)?;
```

Comment:
(nit): `proxy_ip` and `resolved_ip` might be better names for these to avoid confusion with internal IPs referring to a customer's VPC subnet. That's what I'll use when writing docs explaining how this works.

```suggestion
        let internal_ip = ip_provider.get_proxy_ip_for(external_ip)?;
```